pkg/apis/pingcap/v1alpha1/types.go

```diff
@@ -174,91 +203,150 @@ type TidbClusterStatus struct {
 // +k8s:openapi-gen=true
 // PDSpec contains details of PD members
 type PDSpec struct {
-	// +k8s:openapi-gen=false
-	ComponentSpec `json:",inline"`
-
+	ComponentSpec               `json:",inline"`
 	corev1.ResourceRequirements `json:",inline"`
 
+	// The desired ready replicas
+	// +kubebuilder:validation:Minimum=1
 	Replicas int32 `json:"replicas"`
-	// +k8s:openapi-gen=false
-	Service          *ServiceSpec `json:"service,omitempty"`
-	StorageClassName string       `json:"storageClassName,omitempty"`
 
+	// Base image of the component, image tag is now allowed during validation
+	// +kubebuilder:default=pingcap/pd
+	// +kubebuilder:validation:Pattern=^[^:]+$
```

Comment:
Do not allow `:` character, this is merely disallow image tag in `baseImage`, a complete image regex is too complex to write here.

Reply:
We can't use this regex，the image reference may include the port number, such as `10.0.0.1:5000/pingcap/tikv`. We can refer to [here](https://github.com/docker/distribution/blob/master/reference/regexp.go#L68-L70) to generate the correct regular expression, I tested it works. but the regular expression is too long. This is the correct regular expression `^((?:(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(?:(?:\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))+)?(?::[0-9]+)?/)?[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?(?:(?:/[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?)+)?)$ `

Reply:
Good catch, I'd remove it for now and do it programmatically in the validation webhook